File: examples/output/dotnet/Domain/Company.Product.Domain.UseCases/GetAnimalsUseCase.cs
using Company.Product.Domain.UseCases.Types;

using System.Collections.Generic;
using System.Threading.Tasks;

namespace Company.Product.Domain.UseCases
{
    public interface GetAnimalsUseCase
    {
        Task<IEnumerable<Animal>> GetAnimals(string filter, int limit, int offset);
    }
}

File: reference/springboot/domain-usecases/src/main/java/org/example/domain/usecases/GetAnimalsUseCase.java
package org.example.domain.usecases;

import java.util.List;

import org.example.domain.usecases.exceptions.NotAuthenticatedException;
import org.example.domain.usecases.exceptions.NotAuthorizedException;
import org.example.domain.usecases.types.Animal;

public interface GetAnimalsUseCase {

    List<Animal> getAnimals(int limit, int offset) throws NotAuthenticatedException, NotAuthorizedException;
}
